dts/bindings/ethernet/ethernet-phy.yaml

```diff
@@ -25,3 +25,9 @@ properties:
       - "10BASE-T Full-Duplex"
       - "100BASE-T Half-Duplex"
       - "100BASE-T Full-Duplex"
+  interface:
```

Comment:
it makes more sense to me to keep it in the eth driver because is the MAC which needs this information to most likely setup the interface to the PHY. And now that we are at it, we should probably make it part of `ethernet-controller` as `phy-connection-type`. See Linux binding as a reference: https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/net/ethernet-controller.yaml#L57
Most of the other eth bindings in Zephyr are already using `phy-connection-type`.